### ISSUE TITLE
Received unsolicited notification

### ISSUE THREAD

**Issue by 011V32** [ronengr/hass_nuki_bt#66]
## System Information

version | core-2024.6.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.2
os_name | Linux
os_version | 6.6.33-haos
arch | x86_64
timezone | Europe/Vienna
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.34.0
Stage | running
Available Repositories | 1391
Downloaded Repositories | 11

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 12.4
-- | --
update_channel | stable
supervisor_version | supervisor-2024.06.2
agent_version | 1.6.0
docker_version | 26.1.4
disk_total | 468.7 GB
disk_used | 17.1 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | true
ntp_synchronized | true
virtualization | 
board | generic-x86-64
supervisor_api | ok
version_api | ok
installed_addons | Samba share (12.3.1), File editor (5.8.0), RaspberryMatic CCU (3.75.7.20240601), Advanced SSH & Web Terminal (18.0.0), Mosquitto broker (6.4.1), VNC Viewer (1.6.0), ESPHome (2024.6.4), HomeConnect2MQTT (v0.1.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 3
-- | --
resources | 0
views | 14
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | June 21, 2024 at 11:19
-- | --
current_recorder_run | June 28, 2024 at 14:47
estimated_db_size | 297.54 MiB
database_engine | sqlite
database_version | 3.44.2

</details>

<details><summary>Solcast PV Forecast</summary>

can_reach_server | ok
-- | --
used_requests | 2
rooftop_site_count | 1

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/ronengr/hass_nuki_bt/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Nuki Device type

Nuki Smart Lock 3

### hass_nuki_bt version

v0.0.14

### Describe the issue

I have also integrated the lock into HomeKit. In HomeKit I have an automation that locks the lock at 10 p.m.
Whenever this automation is carried out I get errors.
`Logger: pyNukiBT.nuki
Source: runner.py:190
First occurred: June 28, 2024 at 22:00:12 (4 occurrences)
Last logged: June 29, 2024 at 22:00:10

HomeAssistant: Received unsolicited notification: Container: auth_id = b'\x04\x00\xf0\t' (total 4) command = (enum) CHALLENGE 4 payload = Container: nonce = b'\xa5\x07\xe4\xd4\xb0}\xbcw\xb9\xc3\xf5\xc5\x97\xfbR\x0e'... (truncated, total 32) unknown = b'' (total 0) crc = 31944
was expecting CHALLENGE
HomeAssistant: Received unsolicited notification: Container: auth_id = b'\x04\x00\xf0\t' (total 4) command = (enum) CHALLENGE 4 payload = Container: nonce = b'\xb3J\xd3"s\xf3\x7fl\x107\x861\x03\x83\x97\x0e'... (truncated, total 32) unknown = b'' (total 0) crc = 38749`

### Reproduction steps

1. Add lock to HomeKit.
2. Add automation to close in HomeKit.
3. When this automation is executed the error occurs.

### Debug logs

```text
2024-06-28 22:00:02.211 INFO (MainThread) [pyNukiBT.nuki] Lock action LOCK
2024-06-28 22:00:02.211 INFO (MainThread) [pyNukiBT.nuki] sending encrypted command REQUEST_DATA
2024-06-28 22:00:02.212 INFO (MainThread) [pyNukiBT.nuki] Trying to send data. Attempt 1
2024-06-28 22:00:04.147 DEBUG (MainThread) [pyNukiBT.nuki] Services ['00001801-0000-1000-8000-00805f9b34fb (Handle: 1): Generic Attribute Profile', '00001800-0000-1000-8000-00805f9b34fb (Handle: 9): Generic Access Profile', '0000180a-0000-1000-8000-00805f9b34fb (Handle: 14): Gardena DeviceInformation', 'a92ee100-5501-11e4-916c-0800200c9a66 (Handle: 23): Unknown', 'a92ee200-5501-11e4-916c-0800200c9a66 (Handle: 27): Unknown', '00000045-0000-1000-8000-0026bb765291 (Handle: 36): Unknown', '00000096-0000-1000-8000-0026bb765291 (Handle: 53): Unknown', '0000003e-0000-1000-8000-0026bb765291 (Handle: 71): Unknown', '000000a2-0000-1000-8000-0026bb765291 (Handle: 105): Unknown', '00000055-0000-1000-8000-0026bb765291 (Handle: 114): Unknown']
2024-06-28 22:00:04.148 DEBUG (MainThread) [pyNukiBT.nuki] Characteristics ['00002a05-0000-1000-8000-00805f9b34fb (Handle: 3): Service Changed', '00002b2a-0000-1000-8000-00805f9b34fb (Handle: 6): Database Hash', '00002b29-0000-1000-8000-00805f9b34fb (Handle: 8): Client Supported Features', '00002a00-0000-1000-8000-00805f9b34fb (Handle: 11): Device Name', '00002a01-0000-1000-8000-00805f9b34fb (Handle: 13): Appearance', '00002a25-0000-1000-8000-00805f9b34fb (Handle: 16): Serial Number String', '00002a27-0000-1000-8000-00805f9b34fb (Handle: 18): Hardware Revision String', '00002a26-0000-1000-8000-00805f9b34fb (Handle: 20): Gardena DeviceInformation Firmware Version', '00002a24-0000-1000-8000-00805f9b34fb (Handle: 22): Gardena DeviceInformation Model Number', 'a92ee101-5501-11e4-916c-0800200c9a66 (Handle: 25): Unknown', 'a92ee201-5501-11e4-916c-0800200c9a66 (Handle: 29): Unknown', 'a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown', 'a92ee203-5501-11e4-916c-0800200c9a66 (Handle: 35): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 38): Unknown', '000000a5-0000-1000-8000-0026bb765291 (Handle: 40): Unknown', '0000001d-0000-1000-8000-0026bb765291 (Handle: 43): Unknown', '0000001e-0000-1000-8000-0026bb765291 (Handle: 47): MCAP Control Channel', '00000023-0000-1000-8000-0026bb765291 (Handle: 51): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 55): Unknown', '00000079-0000-1000-8000-0026bb765291 (Handle: 57): Unknown', '00000068-0000-1000-8000-0026bb765291 (Handle: 61): Unknown', '0000008f-0000-1000-8000-0026bb765291 (Handle: 65): Unknown', '00000023-0000-1000-8000-0026bb765291 (Handle: 69): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 73): Unknown', '00000014-0000-1000-8000-0026bb765291 (Handle: 75): Hardcopy Data Channel', '00000020-0000-1000-8000-0026bb765291 (Handle: 78): Unknown', '00000021-0000-1000-8000-0026bb765291 (Handle: 81): Unknown', '00000023-0000-1000-8000-0026bb765291 (Handle: 84): Unknown', '00000030-0000-1000-8000-0026bb765291 (Handle: 87): Unknown', '00000052-0000-1000-8000-0026bb765291 (Handle: 90): Unknown', '00000053-0000-1000-8000-0026bb765291 (Handle: 93): Unknown', '00000037-0000-1000-8000-0026bb765291 (Handle: 96): Unknown', '00000220-0000-1000-8000-0026bb765291 (Handle: 99): Unknown', '000000a6-0000-1000-8000-0026bb765291 (Handle: 102): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 107): Unknown', '000000a5-0000-1000-8000-0026bb765291 (Handle: 109): Unknown', '00000037-0000-1000-8000-0026bb765291 (Handle: 112): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 116): Unknown', '0000004c-0000-1000-8000-0026bb765291 (Handle: 118): Unknown', '0000004e-0000-1000-8000-0026bb765291 (Handle: 121): Unknown', '0000004f-0000-1000-8000-0026bb765291 (Handle: 124): Unknown', '00000050-0000-1000-8000-0026bb765291 (Handle: 127): Unknown']
2024-06-28 22:00:04.196 INFO (MainThread) [pyNukiBT.nuki] Connected
2024-06-28 22:00:04.196 INFO (MainThread) [pyNukiBT.nuki] Sending data to Nuki
2024-06-28 22:00:04.269 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'B\x06\xa1\xcc\xe49T\xe3)7\xe5N\x8e]\xf4\x83\x0c\xed\x13ly:\xae\x07\x04\x00\xf0\t8\x00\xc0n\x11~\xe5M\xcf\xf7\xe9\x84*K\x19\x1d\x97\xe4\xec0&,\xbf\xa4\xa9G\xe3\xaf\xe1\x9d\x9d\x9ano\xe5\t\xb0\x0b\x04[\xd78kzUN-B\xeb\xe3D\x15f\x0e\xd3\xb9)\x0c')
2024-06-28 22:00:04.275 INFO (MainThread) [pyNukiBT.nuki] Data sent on attempt 1
2024-06-28 22:00:04.276 INFO (MainThread) [pyNukiBT.nuki] sending encrypted command LOCK_ACTION
2024-06-28 22:00:04.276 INFO (MainThread) [pyNukiBT.nuki] Trying to send data. Attempt 1
2024-06-28 22:00:04.276 INFO (MainThread) [pyNukiBT.nuki] Connected
2024-06-28 22:00:04.276 INFO (MainThread) [pyNukiBT.nuki] Sending data to Nuki
2024-06-28 22:00:04.313 INFO (MainThread) [pyNukiBT.nuki] Data sent on attempt 1
2024-06-28 22:00:04.337 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\xdd[\xa9_\xf2\xd6\xa6\xfb8\x18\x00_!\xa6Cx\xed\xeb*V"\xcc\x92\xfc\x04\x00\xf0\t\x19\x00\x1b\xcfQ?S\xac\x86\xc6$\x19k\xe9`Z\xf05\x1a\xc9\xab\xf6oM^Uf')
2024-06-28 22:00:04.338 DEBUG (MainThread) [pyNukiBT.nuki] Last action: ACCEPTED
2024-06-28 22:00:04.339 INFO (MainThread) [pyNukiBT.nuki] ACCEPTED
2024-06-28 22:00:04.360 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\x1c]Qy\x04\xd4W\xed\xf9x\xe4\xcf\x84\xed\xde\nO\x03\xac=\xba\n\x0f\xc7\x04\x00\xf0\t2\x00\x1e\x7f)\x1e\xb0\xe8#o\xc3\xacW]\x90\x87\xfa\xf8U\xbb\x8e\xba\x94^\x04Yn\x81\xf9\xb9;\x99m\xed35\x04\xe1\xfbz\xe7\xbc3wv\xeb\xbf\xb8a\xd19(')
2024-06-28 22:00:04.362 DEBUG (MainThread) [pyNukiBT.nuki] State: Container:
nuki_state = (enum) DOOR_MODE 2
lock_state = (enum) LOCKING 4
trigger = (enum) SYSTEM 0
current_time = Container:
year = 2024
month = 6
day = 28
hour = 19
minute = 59
second = 50
timezone_offset = 0
critical_battery_state = 124
config_update_count = 16
lock_n_go_timer = 0
last_lock_action = (enum) LOCK 2
last_lock_action_trigger = (enum) SYSTEM 0
last_lock_action_completion_status = (enum) MOTOR_BLOCKED 1
door_sensor_state = (enum) UNAVAILABLE 0
nightmode_active = 0
accessory_battery_state = 0
2024-06-28 22:00:04.362 DEBUG (MainThread) [pyNukiBT.nuki] HomeAssistant: Fire callbacks
2024-06-28 22:00:12.347 DEBUG (MainThread) [pyNukiBT.nuki] Nuki: 54:D2:72:75:B2:FC, RSSI: -56
2024-06-28 22:00:12.348 INFO (MainThread) [pyNukiBT.nuki] Querying Nuki state
2024-06-28 22:00:12.349 INFO (MainThread) [pyNukiBT.nuki] sending encrypted command REQUEST_DATA
2024-06-28 22:00:12.349 INFO (MainThread) [pyNukiBT.nuki] Trying to send data. Attempt 1
2024-06-28 22:00:12.369 DEBUG (MainThread) [pyNukiBT.nuki] Services ['00001801-0000-1000-8000-00805f9b34fb (Handle: 1): Generic Attribute Profile', '00001800-0000-1000-8000-00805f9b34fb (Handle: 9): Generic Access Profile', '0000180a-0000-1000-8000-00805f9b34fb (Handle: 14): Gardena DeviceInformation', 'a92ee100-5501-11e4-916c-0800200c9a66 (Handle: 23): Unknown', 'a92ee200-5501-11e4-916c-0800200c9a66 (Handle: 27): Unknown', '00000045-0000-1000-8000-0026bb765291 (Handle: 36): Unknown', '00000096-0000-1000-8000-0026bb765291 (Handle: 53): Unknown', '0000003e-0000-1000-8000-0026bb765291 (Handle: 71): Unknown', '000000a2-0000-1000-8000-0026bb765291 (Handle: 105): Unknown', '00000055-0000-1000-8000-0026bb765291 (Handle: 114): Unknown']
2024-06-28 22:00:12.370 DEBUG (MainThread) [pyNukiBT.nuki] Characteristics ['00002a05-0000-1000-8000-00805f9b34fb (Handle: 3): Service Changed', '00002b2a-0000-1000-8000-00805f9b34fb (Handle: 6): Database Hash', '00002b29-0000-1000-8000-00805f9b34fb (Handle: 8): Client Supported Features', '00002a00-0000-1000-8000-00805f9b34fb (Handle: 11): Device Name', '00002a01-0000-1000-8000-00805f9b34fb (Handle: 13): Appearance', '00002a25-0000-1000-8000-00805f9b34fb (Handle: 16): Serial Number String', '00002a27-0000-1000-8000-00805f9b34fb (Handle: 18): Hardware Revision String', '00002a26-0000-1000-8000-00805f9b34fb (Handle: 20): Gardena DeviceInformation Firmware Version', '00002a24-0000-1000-8000-00805f9b34fb (Handle: 22): Gardena DeviceInformation Model Number', 'a92ee101-5501-11e4-916c-0800200c9a66 (Handle: 25): Unknown', 'a92ee201-5501-11e4-916c-0800200c9a66 (Handle: 29): Unknown', 'a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown', 'a92ee203-5501-11e4-916c-0800200c9a66 (Handle: 35): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 38): Unknown', '000000a5-0000-1000-8000-0026bb765291 (Handle: 40): Unknown', '0000001d-0000-1000-8000-0026bb765291 (Handle: 43): Unknown', '0000001e-0000-1000-8000-0026bb765291 (Handle: 47): MCAP Control Channel', '00000023-0000-1000-8000-0026bb765291 (Handle: 51): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 55): Unknown', '00000079-0000-1000-8000-0026bb765291 (Handle: 57): Unknown', '00000068-0000-1000-8000-0026bb765291 (Handle: 61): Unknown', '0000008f-0000-1000-8000-0026bb765291 (Handle: 65): Unknown', '00000023-0000-1000-8000-0026bb765291 (Handle: 69): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 73): Unknown', '00000014-0000-1000-8000-0026bb765291 (Handle: 75): Hardcopy Data Channel', '00000020-0000-1000-8000-0026bb765291 (Handle: 78): Unknown', '00000021-0000-1000-8000-0026bb765291 (Handle: 81): Unknown', '00000023-0000-1000-8000-0026bb765291 (Handle: 84): Unknown', '00000030-0000-1000-8000-0026bb765291 (Handle: 87): Unknown', '00000052-0000-1000-8000-0026bb765291 (Handle: 90): Unknown', '00000053-0000-1000-8000-0026bb765291 (Handle: 93): Unknown', '00000037-0000-1000-8000-0026bb765291 (Handle: 96): Unknown', '00000220-0000-1000-8000-0026bb765291 (Handle: 99): Unknown', '000000a6-0000-1000-8000-0026bb765291 (Handle: 102): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 107): Unknown', '000000a5-0000-1000-8000-0026bb765291 (Handle: 109): Unknown', '00000037-0000-1000-8000-0026bb765291 (Handle: 112): Unknown', 'e604e95d-a759-4817-87d3-aa005083a0d1 (Handle: 116): Unknown', '0000004c-0000-1000-8000-0026bb765291 (Handle: 118): Unknown', '0000004e-0000-1000-8000-0026bb765291 (Handle: 121): Unknown', '0000004f-0000-1000-8000-0026bb765291 (Handle: 124): Unknown', '00000050-0000-1000-8000-0026bb765291 (Handle: 127): Unknown']
2024-06-28 22:00:12.372 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\xce\xea\xe2\x1c\xe0\xf7$\xd5\x96\xea\xf2\x1f\x8d7S\x1b\xa8QnC\x8a\x18\x9b\xde\x04\x00\xf0\t2\x00\xb8\x14\xc3\n\x12\\\xff\xcd3v\xa4\x8d\xa5\x10P\xcc\x978\xf78I\xf5)mP\x8b\x1fy\xdb\xb6D[\xb0\x930\x03q.\xcd\xe2##\x86\xda\xc6\xe3Ss7\x1d')
2024-06-28 22:00:12.397 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\xe7\x8d\x96\x988/$\xe66\xf3\x98\x10\xac\x0b\xde\x0f\x04 \x8f\xc7\xa0\xd2\x1fK\x04\x00\xf0\t\x19\x001\xb1\xdc\x062\xcfd\xa0\xcb\x92F\xc4\xa1r\xc9-\xcb\x00\x88\xacA\xf7O\xecB')
2024-06-28 22:00:12.398 DEBUG (MainThread) [pyNukiBT.nuki] Last action: COMPLETED
2024-06-28 22:00:12.398 INFO (MainThread) [pyNukiBT.nuki] Last action: COMPLETED
2024-06-28 22:00:12.398 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\xe7\x8d\x96\x988/$\xe66\xf3\x98\x10\xac\x0b\xde\x0f\x04 \x8f\xc7\xa0\xd2\x1fK\x04\x00\xf0\t\x19\x001\xb1\xdc\x062\xcfd\xa0\xcb\x92F\xc4\xa1r\xc9-\xcb\x00\x88\xacA\xf7O\xecB')
2024-06-28 22:00:12.399 DEBUG (MainThread) [pyNukiBT.nuki] Last action: COMPLETED
2024-06-28 22:00:12.399 INFO (MainThread) [pyNukiBT.nuki] Last action: COMPLETED
2024-06-28 22:00:12.424 INFO (MainThread) [pyNukiBT.nuki] Connected
2024-06-28 22:00:12.424 INFO (MainThread) [pyNukiBT.nuki] Sending data to Nuki
2024-06-28 22:00:12.478 DEBUG (MainThread) [pyNukiBT.nuki] Nuki: 54:D2:72:75:B2:FC, RSSI: -56
2024-06-28 22:00:12.478 INFO (MainThread) [pyNukiBT.nuki] Ignoring duplicate beacon from Nuki 54:D2:72:75:B2:FC
2024-06-28 22:00:12.497 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\xf5\xd15.\xa1\x00u\xe9*t3M\x979%~\xc3d3\x84\xe663\xd9\x04\x00\xf0\t2\x00\x07BeA\x0c\x0eF\x1a\x9c%E>rq\xd5\xd0p=Q\x9f5\xbb\xa5\xba\x8c,\xd2J\x84@\xaa{\xf2\x1e\x12\x8b|\xa6\x9b5\xd0U\t\xfcC-e\x8aie')
2024-06-28 22:00:12.499 DEBUG (MainThread) [pyNukiBT.nuki] State: Container:
nuki_state = (enum) DOOR_MODE 2
lock_state = (enum) LOCKED 1
trigger = (enum) SYSTEM 0
current_time = Container:
year = 2024
month = 6
day = 28
hour = 19
minute = 59
second = 58
timezone_offset = 0
critical_battery_state = 124
config_update_count = 16
lock_n_go_timer = 0
last_lock_action = (enum) LOCK 2
last_lock_action_trigger = (enum) SYSTEM 0
last_lock_action_completion_status = (enum) SUCCESS 0
door_sensor_state = (enum) UNAVAILABLE 0
nightmode_active = 0
accessory_battery_state = 0
2024-06-28 22:00:12.499 DEBUG (MainThread) [pyNukiBT.nuki] HomeAssistant: Fire callbacks
2024-06-28 22:00:12.502 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\xf5\xd15.\xa1\x00u\xe9*t3M\x979%~\xc3d3\x84\xe663\xd9\x04\x00\xf0\t2\x00\x07BeA\x0c\x0eF\x1a\x9c%E>rq\xd5\xd0p=Q\x9f5\xbb\xa5\xba\x8c,\xd2J\x84@\xaa{\xf2\x1e\x12\x8b|\xa6\x9b5\xd0U\t\xfcC-e\x8aie')
2024-06-28 22:00:12.503 DEBUG (MainThread) [pyNukiBT.nuki] State: Container:
nuki_state = (enum) DOOR_MODE 2
lock_state = (enum) LOCKED 1
trigger = (enum) SYSTEM 0
current_time = Container:
year = 2024
month = 6
day = 28
hour = 19
minute = 59
second = 58
timezone_offset = 0
critical_battery_state = 124
config_update_count = 16
lock_n_go_timer = 0
last_lock_action = (enum) LOCK 2
last_lock_action_trigger = (enum) SYSTEM 0
last_lock_action_completion_status = (enum) SUCCESS 0
door_sensor_state = (enum) UNAVAILABLE 0
nightmode_active = 0
accessory_battery_state = 0
2024-06-28 22:00:12.503 DEBUG (MainThread) [pyNukiBT.nuki] HomeAssistant: Fire callbacks
2024-06-28 22:00:12.505 INFO (MainThread) [pyNukiBT.nuki] Data sent on attempt 1
2024-06-28 22:00:12.505 DEBUG (MainThread) [pyNukiBT.nuki] State: Container:
nuki_state = (enum) DOOR_MODE 2
lock_state = (enum) LOCKED 1
trigger = (enum) SYSTEM 0
current_time = Container:
year = 2024
month = 6
day = 28
hour = 19
minute = 59
second = 58
timezone_offset = 0
critical_battery_state = 124
config_update_count = 16
lock_n_go_timer = 0
last_lock_action = (enum) LOCK 2
last_lock_action_trigger = (enum) SYSTEM 0
last_lock_action_completion_status = (enum) SUCCESS 0
door_sensor_state = (enum) UNAVAILABLE 0
nightmode_active = 0
accessory_battery_state = 0
2024-06-28 22:00:12.505 INFO (MainThread) [pyNukiBT.nuki] request 1 log entries, start=0
2024-06-28 22:00:12.505 INFO (MainThread) [pyNukiBT.nuki] sending encrypted command REQUEST_DATA
2024-06-28 22:00:12.506 INFO (MainThread) [pyNukiBT.nuki] Trying to send data. Attempt 1
2024-06-28 22:00:12.506 INFO (MainThread) [pyNukiBT.nuki] Connected
2024-06-28 22:00:12.506 INFO (MainThread) [pyNukiBT.nuki] Sending data to Nuki
2024-06-28 22:00:12.558 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\x8f\x12(\xfcE\xc2z\x8f4j\xe2\xa8\xa9$\xdfI\x92\xfb}\xe0\xa3z\x90\xdd\x04\x00\xf0\t8\x00\xe1\xd8\x18\x7f\x99\xdbM\xb9\x93\xba|\xe6-\x8bjw\xdc\x1ek\xf3\x01\t\x86\xaf\xe5\xf0\xa8VY\xa0\x96\xdf6Z\xcaQ\x98\x10\xf1\xc7P\xaa\xe7Y\xae\xd3t\xad\xf5\x91" ?k\xa2\xd5')
2024-06-28 22:00:12.560 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\x8f\x12(\xfcE\xc2z\x8f4j\xe2\xa8\xa9$\xdfI\x92\xfb}\xe0\xa3z\x90\xdd\x04\x00\xf0\t8\x00\xe1\xd8\x18\x7f\x99\xdbM\xb9\x93\xba|\xe6-\x8bjw\xdc\x1ek\xf3\x01\t\x86\xaf\xe5\xf0\xa8VY\xa0\x96\xdf6Z\xcaQ\x98\x10\xf1\xc7P\xaa\xe7Y\xae\xd3t\xad\xf5\x91" ?k\xa2\xd5')
2024-06-28 22:00:12.561 ERROR (MainThread) [pyNukiBT.nuki] HomeAssistant: Received unsolicited notification: Container:
auth_id = b'\x04\x00\xf0\t' (total 4)
command = (enum) CHALLENGE 4
payload = Container:
nonce = b'\xa5\x07\xe4\xd4\xb0}\xbcw\xb9\xc3\xf5\xc5\x97\xfbR\x0e'... (truncated, total 32)
unknown = b'' (total 0)
crc = 31944
2024-06-28 22:00:12.562 ERROR (MainThread) [pyNukiBT.nuki] was expecting CHALLENGE
2024-06-28 22:00:12.565 INFO (MainThread) [pyNukiBT.nuki] Data sent on attempt 1
2024-06-28 22:00:12.565 INFO (MainThread) [pyNukiBT.nuki] sending encrypted command REQUEST_LOG_ENTRIES
2024-06-28 22:00:12.566 INFO (MainThread) [pyNukiBT.nuki] Trying to send data. Attempt 1
2024-06-28 22:00:12.566 INFO (MainThread) [pyNukiBT.nuki] Connected
2024-06-28 22:00:12.566 INFO (MainThread) [pyNukiBT.nuki] Sending data to Nuki
2024-06-28 22:00:12.623 INFO (MainThread) [pyNukiBT.nuki] Data sent on attempt 1
2024-06-28 22:00:12.643 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b"\xe0)\xa1a\xa1\xc2YiL\x18}\xb0\xfd<\xf7\xe5t\xaf5\xe6\x1aT\x1c\xba\x04\x00\xf0\tM\x00\x8b\xe4\x91\xb7\x9f\x11\xa6Sn\xbfj\x06\xa8\xa9\xd7l\xf4k\x8e\x94r7\x1dOd\xb6\xcf\xc4\x05\xa2f\xa8\x00\x8f\x89HtX9\r\xb1\xa8\xf6B\xe8`\x0c\xe2\x0e\xa7b\x1e\x94O}\xbat\xc0\xdd2\xbc\xd9&\'[\xa2\x04\x10\xedStlUt\xa5\xb7=")
2024-06-28 22:00:12.645 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b"\xe0)\xa1a\xa1\xc2YiL\x18}\xb0\xfd<\xf7\xe5t\xaf5\xe6\x1aT\x1c\xba\x04\x00\xf0\tM\x00\x8b\xe4\x91\xb7\x9f\x11\xa6Sn\xbfj\x06\xa8\xa9\xd7l\xf4k\x8e\x94r7\x1dOd\xb6\xcf\xc4\x05\xa2f\xa8\x00\x8f\x89HtX9\r\xb1\xa8\xf6B\xe8`\x0c\xe2\x0e\xa7b\x1e\x94O}\xbat\xc0\xdd2\xbc\xd9&\'[\xa2\x04\x10\xedStlUt\xa5\xb7=")
2024-06-28 22:00:12.666 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\xeep\xa5R\xe2d\xd4=\xfa\x05D\xb0\xa5\x8e\x84\xf9\x87\xe7\xad\xe8\x04\x19he\x04\x00\xf0\t\x19\x00\xef\x8c\x14\x05rr\xed\x1f\xb8d\xe2\xd9\x19\x01\\\\\xef\xd2\xa2\xed\xb3$x2\x87')
2024-06-28 22:00:12.667 DEBUG (MainThread) [pyNukiBT.nuki] Last action: COMPLETED
2024-06-28 22:00:12.667 DEBUG (MainThread) [pyNukiBT.nuki] Notification handler: a92ee202-5501-11e4-916c-0800200c9a66 (Handle: 32): Unknown, data: bytearray(b'\xeep\xa5R\xe2d\xd4=\xfa\x05D\xb0\xa5\x8e\x84\xf9\x87\xe7\xad\xe8\x04\x19he\x04\x00\xf0\t\x19\x00\xef\x8c\x14\x05rr\xed\x1f\xb8d\xe2\xd9\x19\x01\\\\\xef\xd2\xa2\xed\xb3$x2\x87')
2024-06-28 22:00:12.668 DEBUG (MainThread) [pyNukiBT.nuki] Last action: COMPLETED
2024-06-28 22:00:12.668 INFO (MainThread) [pyNukiBT.nuki] Last action: COMPLETED
2024-06-28 22:00:12.668 DEBUG (MainThread) [pyNukiBT.nuki] Container:
status = (enum) COMPLETED 0
2024-06-28 22:00:12.668 DEBUG (MainThread) [pyNukiBT.nuki] [Container(index=2619, timestamp=Container(year=2024, month=6, day=28, hour=19, minute=59, second=58), auth_id=166723588, name=u'HomeAssistant', type=uEnumIntegerString.new(2, 'LOCK_ACTION'), data=Container(lock_action=uEnumIntegerString.new(2, 'LOCK'), trigger=uEnumIntegerString.new(0, 'SYSTEM'), flags=0, completion_status=uEnumIntegerString.new(0, 'SUCCESS'), padding=None)), Container(index=2619, timestamp=Container(year=2024, month=6, day=28, hour=19, minute=59, second=58), auth_id=166723588, name=u'HomeAssistant', type=uEnumIntegerString.new(2, 'LOCK_ACTION'), data=Container(lock_action=uEnumIntegerString.new(2, 'LOCK'), trigger=uEnumIntegerString.new(0, 'SYSTEM'), flags=0, completion_status=uEnumIntegerString.new(0, 'SUCCESS'), padding=None))]
```

**Comment by ronengr** [comment]
@011V32 
Other then the log, is there any issue?
I did had some issues with logging in this area, I hope to fix them in the next version.
But as i dont have homekit, I have no way to test/reproduce this issue.

**Comment by 011V32** [comment]
@ronengr 
I don't think so. I have to admit that I only use it sometimes through automations and I haven't noticed anything so far.
I can observe it again when your changes are there. 👍 

But I have another behavior:
If an action is carried out that has actually already been completed, then the status is toggled 2 times.
So if the status is already open and an automation "open" is triggert again, then it is briefly shown as closed, even though only opening is requested.
The same applies the other way around.
This means I always get two notifications even though the status doesn't actually change.

**Comment by ronengr** [comment]
@011V32 the version i released yesterday (0.0.15 and up) already has the fix I was talking about.

Regarding the status change you described - I didn't encounter such issue, even though i have several automatons that open the door without checking it's status (so sending "open" when it is already open).
Can you send me logs so i can see what exactly is happening?
Could it be that it changes from locked->locking->locked and not from locked->unlocked->locked? if this is the case, it makes more sense.

**Comment by 011V32** [comment]
@ronengr
It didn't show up for me until I initiate the update information manually.
I'm now on 0.0.16 and will report back. 👍

**Comment by 011V32** [comment]
@ronengr 
> Regarding the status change you described - I didn't encounter such issue, even though i have several automatons that open the door without checking it's status (so sending "open" when it is already open).
> Can you send me logs so i can see what exactly is happening?
> Could it be that it changes from locked->locking->locked and not from locked->unlocked->locked? if this is the case, it makes more sense.

I think with version 0.0.16, the system no longer switches back and forth twice when the same state is triggered again.

With version 0.0.14 the Logbook was as follow wenn opening with open lock (from bottom to up like in HA):
> Nuki_3575B2FC Last action user name changed to HomeAssistant
> 14:05:04 - 24 minutes ago
> Nuki_3575B2FC Lock state changed to UNLOCKED
> 14:05:04 - 24 minutes ago
> Nuki_3575B2FC Lock was unlocked triggered by HomeKit send command unlock for Lock (Nuki_3575B2FC Lock)
> 14:05:04 - 24 minutes ago
> Nuki_3575B2FC Last Action Trigger changed to SYSTEM
> 14:05:03 - 24 minutes ago
> Nuki_3575B2FC Last lock action changed to UNLOCK
> 14:05:03 - 24 minutes ago
> Nuki_3575B2FC Lock state changed to UNLOCKING
> 14:05:03 - 24 minutes ago
> Nuki_3575B2FC Lock is unlocking triggered by HomeKit send command unlock for Lock (Nuki_3575B2FC Lock)
> 14:05:03 - 24 minutes ago

With version 0.0.16 it ist now:
> Nuki_3575B2FC Last Nuki command status changed to COMPLETED
> 14:27:03 - 4 minutes ago
> Nuki_3575B2FC Last Nuki command status changed to ACCEPTED
> 14:27:03 - 4 minutes ago

I will test a little bit 👍

**Comment by 011V32** [comment]
@ronengr 
I think that the state only needs to change if it can change. I can only press Unlock in the UI when it is locked and not when it is already unlocked.
So if you execute Unlock and it is already open, then you will get instant the status succesful back.
But I have to admit that I don't know the exact topic behind HA and the other systems. I can only express my opinion from a user perspective.

**Comment by ronengr** [comment]
I an ideal world you are probably correct. But as there are communication issues, the "real" state of the lock is not always updated correctly in HA. (Sometimes a status change update is missed, for several reasons i cannot control).
This is why I don't want to block commands at the integration level, but rather prefer to send every command to the lock itself, regardless of the state i think the lock is in at the moment.

The question is mostly from user perspective - what do you (or the average user) prefer:
1. Get an immediate indication when you execute a lock/unlock command (change to locking/unlocking), and than getting the real state from the lock after HA was able to communicate with the Nuki lock.
2. Don't get an immediate indication, only get an indication after HA was able to communicate with Nuki lock.

Personally I'm not sure which is better. Should we indicate locking/unlocking only when the lock is actually turning (this is the state now), or should we indicate locking/unlocking from the moment we start sending the relevant command for the lock.

**Comment by 011V32** [comment]
For me it looks like this:
![image](https://github.com/user-attachments/assets/c4f9c764-2e40-4228-aa3a-e5c5e59052a9)

**Comment by ronengr** [comment]
I'm not familiar with HomeKit, so i wouldn't know why it acts this way.

Anyway, I will probably keep the current state (updating locking status only after reading from the lock)

**Comment by ronengr** [comment]
@011V32 
I'm not sure i fully understand the issue in the screenshot you provided.
It seems locking was triggered by HomeKit once (at 22:00:03) and was finished only once.
What am I missing?

Anyway, I think you are correct and it is better to open a new issue for this.
